Source/FieldSolver/WarpXPushFieldsEM.cpp

```diff
@@ -68,6 +68,11 @@ namespace {
         solver.ForwardTransform(*current[2], Idx::Jz);
         solver.ForwardTransform(*rho, Idx::rho_old, 0);
         solver.ForwardTransform(*rho, Idx::rho_new, 1);
+#ifdef WARPX_DIM_RZ
+        if (WarpX::use_kspace_filter) {
+            solver.ApplyFilter(Idx::Jx, Idx::Jy, Idx::Jz);
```

Comment:
Is it necessary to apply the filter to `rho_old` and `rho_new` as well? I am asking because they are also both used to push the electric field when we call `solver.pushSpectralFields();`:
https://github.com/ECP-WarpX/WarpX/blob/af7144301bf0d6bbc2263ef389c321cb0cb6fe40/Source/FieldSolver/SpectralSolver/SpectralAlgorithms/PsatdAlgorithmRZ.cpp#L120-L129

Reply:
Yes, you are correct. This was an oversight. I'll make the change.